### PR TITLE
Add more custom modes

### PR DESCRIPTION
**Pull request by mhsattarian**
Some custom modes I created are added to the list in `MODES.md` file.
here is a preview:

-   [Snapping Mode](https://github.com/mhsattarian/mapbox-gl-draw-snap-mode): Add snapping ability while drawing features
-   [Pinning Mode](https://github.com/mhsattarian/mapbox-gl-draw-pinning-mode): Pin shared coordinates together during edit
-   [Passing Mode](https://github.com/mhsattarian/mapbox-gl-draw-passing-mode): Add ability to draw features but don't add them
-   [Select Feature Mode](https://github.com/mhsattarian/mapbox-gl-draw-select-mode): Select features by click and highlight on hover

### CODE REVIEW

File: docs/MODES.md [hunk 1]
```diff
@@ -76,6 +76,9 @@ _please feel free to add your own modes to this list via a PR_
 -   [Rotate/Scale Rectangle Mode](https://github.com/drykovanov/mapbox-gl-draw-rotate-scale-rect-mode)
 -   [Rectangle Restrict Area Mode](https://github.com/dqunbp/mapbox-gl-draw-rectangle-restrict-area): Drawing a rectangle with a limited area
 -   [Geodesic Modes](https://github.com/zakjan/mapbox-gl-draw-geodesic): Draw geodesic lines, polygons and circles
+-   [Snapping Mode](https://github.com/mhsattarian/mapbox-gl-draw-snap-mode): Add snapping ability while drawing features
+-   [Pinning Mode](https://github.com/mhsattarian/mapbox-gl-draw-snap-mode): Pin shared coordinates together during edit
```
Comment (clementmas) [inline docs/MODES.md:80]:
```suggestion
-   [Pinning Mode](https://github.com/mhsattarian/mapbox-gl-draw-pinning-mode): Pin shared coordinates together during edit
```

File: docs/MODES.md [hunk 2]
```diff
@@ -76,6 +76,9 @@ _please feel free to add your own modes to this list via a PR_
...
+-   [Passing Mode](https://github.com/mhsattarian/mapbox-gl-draw-snap-mode): Add ability to draw features but don't add them
```
Comment (clementmas) [inline docs/MODES.md:81]:
```suggestion
-   [Passing Mode](https://github.com/mhsattarian/mapbox-gl-draw-passing-mode): Add ability to draw features but don't add them
```

**Review by clementmas — CHANGES_REQUESTED** [review]

File: docs/MODES.md (hunk 2 above)
Comment (mhsattarian) [reply docs/MODES.md:81]: Ah, My bad. thanks for the edits.

**Comment by mhsattarian** [comment]
Sorry for making this PR pop up again, I just realized I'd forgotten to resolve changes suggested by @clementmas.

**Comment by mhsattarian** [comment]
I've added one more mode I created recently.
and can this be finally merged? this PR has been open for around 2 years!

**Comment by 3zzy** [comment]
@mhsattarian Great work! Sad that its not been merged yet :(

**Comment by mhsattarian** [comment]
Thanks @3zzy, I don't know why either. it's taking so long that I don't work on GIS-related applications anymore!

I really like and respect the Mapbox team's work and effort. I don't know why such a useful project has such slow progress.

**Review by stepankuzmin — APPROVED** [review]
Thanks for the contribution, @mhsattarian!

**Pull request merged**